src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -266,7 +284,10 @@ public void testSerialization() throws IOException {
      * @return a new {@link QueryParseContext} based on the base test index and queryParserService
      */
     protected static QueryParseContext createContext() {
-        return new QueryParseContext(index, queryParserService);
+        QueryParseContext context = new QueryParseContext(index, queryParserService);
+        QueryParseContext.setTypes(currentTypes);
```

Comment:
mmm don't we have this somewhere else already? being it static, we shouldn't do it every time we create a new context, only once